src/ert/_c_wrappers/enkf/ert_config.py

```diff
@@ -644,6 +644,7 @@ def _workflows_from_dict(
                     work[0],
                     substitution_list,
                     workflow_jobs,
+                    use_new_parser=USE_NEW_PARSER_BY_DEFAULT,
```

Comment:
This should be set to the same as the `use_new_parser` parameter and not the constant.